pkg/client/powervs_client.go

```diff
@@ -292,37 +306,79 @@ func (p *powerVSClient) GetInstances() (*models.PVMInstances, error) {
 	return p.InstanceClient.GetAll()
 }
 
-func (p *powerVSClient) GetCloudServiceInstances() ([]bluemixmodels.ServiceInstanceV2, error) {
-	var instances []bluemixmodels.ServiceInstanceV2
-	svcs, err := p.ResourceClient.ListInstances(controllerv2.ServiceInstanceQuery{
-		Type: "service_instance",
-	})
-	if err != nil {
-		return svcs, fmt.Errorf("failed to list the service instances: %v", err)
-	}
-	for _, svc := range svcs {
-		if svc.Crn.ServiceName == PowerServiceType {
-			instances = append(instances, svc)
+func (p *powerVSClient) GetCloudServiceInstances() ([]resourcecontrollerv2.ResourceInstance, error) {
+	var serviceInstancesList []resourcecontrollerv2.ResourceInstance
+	f := func(start string) (bool, string, error) {
+		listServiceInstanceOptions := &resourcecontrollerv2.ListResourceInstancesOptions{Type: &serviceInstanceType}
+		if start != "" {
+			listServiceInstanceOptions.Start = &start
+		}
+
+		serviceInstances, _, err := p.ResourceClient.ListResourceInstances(listServiceInstanceOptions)
+		if err != nil {
+			return false, "", err
+		}
+		if serviceInstances != nil {
+			serviceInstancesList = append(serviceInstancesList, serviceInstances.Resources...)
+			nextURL, err := serviceInstances.GetNextStart()
+			if err != nil {
+				return false, "", err
+			}
+			return false, *nextURL, nil
 		}
+		return true, "", nil
 	}
-	return instances, nil
-}
 
-func (p *powerVSClient) GetCloudServiceInstanceByName(name string) ([]bluemixmodels.ServiceInstanceV2, error) {
-	var instances []bluemixmodels.ServiceInstanceV2
-	svcs, err := p.ResourceClient.ListInstances(controllerv2.ServiceInstanceQuery{
-		Type: "service_instance",
-		Name: name,
-	})
-	if err != nil {
-		return svcs, fmt.Errorf("failed to list the service instances: %v", err)
+	if err := utils.PagingHelper(f); err != nil {
+		return nil, fmt.Errorf("error listing loadbalancer %v", err)
+	}
+
+	if serviceInstancesList == nil {
+		return nil, errors.New("no service instance is retrieved")
 	}
-	for _, svc := range svcs {
-		if svc.Crn.ServiceName == PowerServiceType {
-			instances = append(instances, svc)
+	return serviceInstancesList, nil
+}
+
+func (p *powerVSClient) GetCloudServiceInstanceByName(name string) (*resourcecontrollerv2.ResourceInstance, error) {
+	var serviceInstancesList []resourcecontrollerv2.ResourceInstance
+	f := func(start string) (bool, string, error) {
+		listServiceInstanceOptions := &resourcecontrollerv2.ListResourceInstancesOptions{Type: &serviceInstanceType, Name: &name}
```

Comment:
what if two different kind of services have same name? can we also restrict based on the catalog type?